src/appleseed-max-impl/appleseedrenderer/appleseedrenderer.cpp

```diff
@@ -1216,14 +1216,14 @@ ParamBlockDesc2 g_param_block_desc(
         p_accessor, &g_pblock_accessor,
     p_end,
 
-    ParamIdNoiseSeed, L"noise_seed", TYPE_INT, P_TRANSIENT, 0,
+    ParamIdPerFrameNoiseSeedVariation, L"noise_seed", TYPE_INT, P_TRANSIENT, 0,
         p_ui, ParamMapIdImageSampling, TYPE_SPINNER, EDITTYPE_INT, IDC_TEXT_NOISE_SEED, IDC_SPINNER_NOISE_SEED, SPIN_AUTOSCALE,
         p_default, 0,
-        p_range, 0, 2000000000,
+        p_range, 0, 2147483647,
```

Comment:
Minimum value should be -2147483648 if possible.

Reply:
I leave the lower value 0. Visual Studio complained when I set the lower value to -2,147,483,64. The range from 0 on upwards is really more than enough for any realistic animation ever to be rendered with appleseed-Max. Max does not recognize uint types in some ui-parameter block settings like p_range, only integer type but the render setting parameter it refers to can be both int or uint.

Reply:
It's not so much about the number of possible values, obviously 2 billion values is more than enough. It's about being able to match any valid noise seed value, and about not introducing artificial limits.

You indeed cannot use -2147483648 directly in C/C++, even with `L` suffix: `-2147483648L`. You would need two L suffixes: `-2147483648LL`, but that would make the constant a `long long` (64-bit).

The cleanest option is to use `numeric_limits<int>::min()` and `numeric_limits<int>::max()` (you'll need to include `<limits>`).

If you look at how they are defined, they use C's `INT_MIN` and `INT_MAX` macros, where `INT_MIN` is typically defined as `(-2147483647 - 1)`.

Reply:
I was curious for why one can't directly write -2147483648 in C, here are some answers:

http://brnz.org/hbr/?p=1451
https://stackoverflow.com/a/9941519/393756

Reply:
Interesting, the reason why VC++ has INT_MIN defined that way. 
But I keep the lower bound to 0. I just checked why negative values don't work and saw that the noise seed parameter has been defined in appleseed core as a 32bit unsigned int in `src/appleseed/renderer/modeling/frame/frame.cpp` line 932 for instance:
`impl->m_noise_seed = m_params.get_optional<uint32>("noise_seed", 0);`

Reply:
In appleseed core it's an unsigned 32-bit value. If you set the noise seed to a negative value, it will simply wrap around. All 32 bits are preserved.

With the range [0, 2147483647] you effectively limit the choice to a 31-bit value.

Reply:
Ok, I allow now the full int range, tested the wraparound and it works.